src/page-multi-facility/ReferenceDataModal/SyncNewReferenceData.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+
+import { referenceFacilitiesProp } from "../../database";
+import { useFacilities } from "../../facilities-context";
+import useScenario from "../../scenario-context/useScenario";
+import { Facility, ModelInputs, ReferenceFacility } from "../types";
+import ReferenceDataModal from ".";
+import {
+  getUnmappedFacilities,
+  getUnmappedReferenceFacilities,
+  ReferenceFacilitySelect,
+  ReferenceFacilitySelections,
+  TitleContainer,
+  TitleText,
+} from "./shared";
+
+const Spacer = styled.span`
+  margin-right: 2em;
+`;
+
+const Title: React.FC<Pick<Props, "stateName" | "systemType">> = ({
+  stateName,
+  systemType,
+}) => (
+  <TitleContainer>
+    <TitleText>Prepopulate Data</TitleText>
+    <TitleText>
+      We've found new facility data - select a corresponding facility to
+      autofill with case data
+    </TitleText>
+    <TitleText>
+      State: {stateName || ""}
+      <Spacer />
+      Type of System: {systemType || ""}
+    </TitleText>
+  </TitleContainer>
+);
+
+interface Props {
+  open: boolean;
+  stateName: ModelInputs["stateName"];
+  systemType: Facility["systemType"];
+  onClose: () => void;
+}
+
+const SyncNewReferenceData: React.FC<Props> = ({
+  open,
+  stateName,
+  systemType,
+  onClose,
+}) => {
+  const [selections, setSelections] = useState<ReferenceFacilitySelections>({});
+  const [scenarioState] = useScenario();
+  const {
+    state: { facilities: facilitiesMapping, referenceFacilities },
+  } = useFacilities();
+  const scenario = scenarioState.data;
+  const mappedReferenceFacilities = scenario?.[referenceFacilitiesProp] || {};
+  const unmappedFacilities = getUnmappedFacilities(
+    mappedReferenceFacilities,
+    facilitiesMapping,
+  );
+  const unmappedReferenceFacilities = getUnmappedReferenceFacilities(
```

Comment:
To clarify my previous comment, we should also return `null` if `unmappedReferenceFacilities` is empty right?

Reply:
Yeah I think that makes sense to add here too, will update this.